packages/core/src/__tests__/recon.test.ts

```diff
@@ -0,0 +1,156 @@
+import { ReconApi, ReconApiConfig } from '../recon.js'
+import { EventID, StreamID } from '@ceramicnetwork/streamid'
+import { FetchRequest, LoggerProvider } from '@ceramicnetwork/common'
+import { jest } from '@jest/globals'
+import { type CAR } from 'cartonne'
+import { toArray, take, lastValueFrom } from 'rxjs'
+
+const RECON_URL = 'http://example.com'
+const LOGGER = new LoggerProvider().getDiagnosticsLogger()
+const MODEL = StreamID.fromString('kjzl6cwe1jw147ww5d8pswh1hjh686mut8v1br10dar8l9a3n1t8l15l0vrzn88')
+const FAKE_EVENT = { id: EventID.createRandom('inmemory', 0), data: {} as CAR }
+
+describe('ReconApi', () => {
+  let mockSendRequest: jest.Mock<FetchRequest>
+  let reconApi: ReconApi
+
+  beforeEach(async () => {
+    mockSendRequest = jest.fn((url: URL | string): Promise<any> => {
+      url = url.toString()
+      if (url.includes('/ceramic/feed/events')) {
+        return Promise.resolve({
+          events: [{ id: EventID.createRandom('inmemory', 0).toString(), data: undefined }],
+          resumeToken: 'test',
+        })
+      }
+      return Promise.resolve()
+    })
+
+    reconApi = new ReconApi(
+      {
+        enabled: true,
+        url: RECON_URL,
+        feedEnabled: true,
+      },
+      LOGGER,
+      mockSendRequest
+    )
+    await reconApi.init()
+    mockSendRequest.mockClear()
+  })
+
+  afterEach(async () => {
+    await reconApi.stop()
+  })
+
+  describe('init', () => {
+    test('should not init if recon is disabled', async () => {
+      const reconApi = new ReconApi(
+        { enabled: false, url: RECON_URL, feedEnabled: true },
+        LOGGER,
+        mockSendRequest
+      )
+      await reconApi.init()
+      expect(mockSendRequest).not.toHaveBeenCalled()
+    })
+
+    test('should not init if already initialized', async () => {
+      // already initialized in beforeEach
+      await reconApi.init()
+
+      expect(mockSendRequest).not.toHaveBeenCalled()
+    })
+
+    test('should not start polling if feed is disabled', async () => {
```

Comment:
I don't quite understand this test.. I'd expect something like the `const received = await lastValueFrom(reconApi.pipe(take(3), toArray()))` call to check if it polled?

Reply:
it says should NOT start polling, so i don't expect to receive anything

But i should try to receive something and then check that I received nothing